plugins/genocide/genocide.cpp

```diff
@@ -16,29 +16,47 @@ class genocide : public bz_Plugin
     virtual void Cleanup ( void );
     virtual void Event ( bz_EventData *eventData );
 
-    bool allowSuicide = false;
-    bool rogueGenocide = false;
+    bool disableSuicide = false;
+    bool rogueAsTeam = false;
 };
 
 BZ_PLUGIN(genocide)
 
-void genocide::Init ( const char* commandLine )
+void genocide::Init ( const char* cmdLine )
 {
     bz_debugMessage(4, "genocide plugin loaded");
-    if (commandLine)
+    if (strlen(cmdLine) != 0)
```

Comment:
This is a fundamental change in functionality/stability.

One of the dangers of pointers is that they have the potential to be null. The previous code tested for null. Here you assume it's valid, and points to memory with a single null byte. 

Testing for an empty string is pretty pointless.